draft-peabody-dispatch-new-uuid-format-03.txt

```diff
@@ -528,1005 +546,889 @@ Internet-Draft               new-uuid-format               February 2022
-   generating these values can assign whatever lengths to each field it
-   deems applicable, as long as it does not break decoding compatibility
-   (i.e.  Unix timestamp (unixts), version (ver) and variant (var) have
-   to stay where they are, and clock sequence counter (seq), random
-   (random) or other implementation specific values must follow the sub-
-   second encoding).
-
-   In Figure 3 the UUIDv7 has been created with millisecond precision
-   with the available sub-second precision bits.
-
-   Examining Figure 3 one can observe:
-
-   *  The first 36 bits have been dedicated to the Unix Timestamp
-      (unixts)
-
-   *  All 12 bits of scenario subsec_a is fully dedicated to millisecond
-      information (msec).
-
-   *  The 4 Version bits remain unchanged (ver).
-
-   *  All 12 bits of subsec_b have been dedicated to a monotonic clock
-      sequence counter (seq).
-
-   *  The 2 Variant bits remain unchanged (var).
-
-   *  Finally the remaining 62 bits in the subsec_seq_node section are
-      layout is filled out with random data to pad the length and
-      provide guaranteed uniqueness (rand).
-
-        0                   1                   2                   3
-        0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
-       +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-       |                            unixts                             |
-       +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-       |unixts |         msec          |  ver  |          seq          |
-       +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-       |var|                         rand                              |
-       +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-       |                             rand                              |
-       +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-
-   Figure 3: UUIDv7 Field and Bit Layout - Encoding Example (Millisecond
-                                 Precision)
-
-
-
-
-
-Peabody & Davis           Expires 5 August 2022                [Page 13]
+   Reliability:
+      Implementations SHOULD use the current timestamp from a reliable
+      source to provide values that are time-ordered and continually
+      increasing.  Care SHOULD be taken to ensure the timestamp does not
+      move backwards which will impact UUID sorting.
+
+   Source:
+      UUID version 1 and 6 both utilize a Gregorian epoch timestamp
+      while UUIDv7 utilizes a Unix Epoch timestamp.  If other timestamp
+      sources or a custom timestamp epoch are required UUIDv8 SHOULD be
+      leveraged.
+
+   Sub-second Precision and Accuracy):
+      Many levels of precision exist for timestamps: milliseconds,
+      microseconds, nanoseconds, and beyond.  Additionally fractional
+      representations of sub-second precision may be desired to create
+      more exotic timestamps.  With UUID version 1 and 6,
```

Comment:
```suggestion
      microseconds, nanoseconds, and beyond.  Additionally fractional
      representations of sub-second precision may be desired to mix
      various levels of precision in a sortable manner.  With UUID version 1 and 6,
```
Fractional representations help inter-operate multiple timestamp precisions. Exotic precision is one efficient mean for that end.